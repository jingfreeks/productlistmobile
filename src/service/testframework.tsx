/* eslint-disable @typescript-eslint/no-explicit-any */
import React,{PropsWithChildren} from 'react';
import {render} from '@testing-library/react-native';
import type {RenderOptions} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import type {AppStore, RootState} from '@/config/store';
import {setupStore} from '@/config/store'
// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({children}: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
