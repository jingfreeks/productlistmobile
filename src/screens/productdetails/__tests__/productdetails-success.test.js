import React from 'react';
import { renderWithProviders } from "../../../service/testframework";
import {Productdetails} from '../index';

jest.mock('../../../service/slice/producSlice', () => ({
    useGetProdDetailsQuery: () => ({
      isLoading: false,
      isSuccess: true,
      isError: false,
      data: {ids: []},
    }),
  }));
describe('Product Details Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Productdetails />);
    expect(all.toJSON()).toMatchSnapshot();
  });
});
