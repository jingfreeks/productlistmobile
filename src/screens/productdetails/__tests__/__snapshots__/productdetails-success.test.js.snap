// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Details Screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      resizeMode="stretch"
      source={
        {
          "uri": undefined,
        }
      }
      style={
        [
          {
            "borderBottomLeftRadius": 10,
            "borderTopLeftRadius": 10,
            "height": 200,
            "marginTop": 10,
            "width": "90%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
            "fontSize": 24,
            "fontWeight": "900",
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0e3f6c",
            "fontFamily": "Avenir",
            "fontSize": 18,
            "fontWeight": "900",
          },
        ]
      }
    >
      Description:
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0e3f6c",
            "fontFamily": "Avenir",
            "fontSize": 18,
            "fontWeight": "900",
          },
        ]
      }
    >
      Category:
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0e3f6c",
            "fontFamily": "Avenir",
            "fontSize": 18,
            "fontWeight": "900",
          },
        ]
      }
    >
      Price:
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
            "fontSize": 12,
            "fontWeight": "400",
          },
        ]
      }
    >
      â‚±
    </Text>
  </View>
</View>
`;
